assigner/backends/base.py

```diff
@@ -0,0 +1,154 @@
+import git
+import re
+from typing import Optional
+
+
+class RepoError(Exception):
```

Comment:
This doesn't have to be done in this PR but I would really like to do some better exception modeling for both GitLab and git--I'd really like to not have command implementations catch and interpret HTTPErrors or GitCommandErrors. Plus, I think that'd make testing easier--we can know what sorts of exceptions a function can throw* and test that Assigner handles them properly.

* except for abstract base classes...I dream of strongly typed algebraic effects but we don't even have those in Haskell yet 🙃

Reply:
This is issue #8 😄